src/components/app/data/hooks/useHasAvailableSubsidiesOrRequests.js

```diff
@@ -0,0 +1,69 @@
+import { useMemo } from 'react';
+import { getOfferExpiringFirst, getPolicyExpiringFirst } from '../../../dashboard/sidebar/utils';
+import { LICENSE_STATUS } from '../../../enterprise-user-subsidy/data/constants';
+import useBrowseAndRequest from './useBrowseAndRequest';
+import useCouponCodes from './useCouponCodes';
+import useEnterpriseOffers from './useEnterpriseOffers';
+import useRedeemablePolicies from './useRedeemablePolicies';
+import useSubscriptions from './useSubscriptions';
+
+function getLearnerCreditSummaryCardData({ enterpriseOffers, redeemableLearnerCreditPolicies }) {
+  const learnerCreditPolicyExpiringFirst = getPolicyExpiringFirst(redeemableLearnerCreditPolicies?.redeemablePolicies);
+  const enterpriseOfferExpiringFirst = getOfferExpiringFirst(enterpriseOffers);
+
+  if (!learnerCreditPolicyExpiringFirst && !enterpriseOfferExpiringFirst) {
+    return undefined;
+  }
+
+  return {
+    expirationDate: (
+      learnerCreditPolicyExpiringFirst?.subsidyExpirationDate || enterpriseOfferExpiringFirst?.endDatetime
+    ),
+  };
+}
+
+/**
+ *
+ * @returns {{
+ * hasAvailableLearnerCreditPolicies: boolean,
+ * hasAssignedCodesOrCodeRequests: boolean,
+ * hasActiveLicenseOrLicenseRequest: boolean,
+ * learnerCreditSummaryCardData: {expirationDate},
+ * hasAvailableSubsidyOrRequests: ({expirationDate}|boolean)
+ * }}
+ */
+export default function useHasAvailableSubsidiesOrRequests() {
+  const { data: subscriptions } = useSubscriptions();
+  const { data: { requests } } = useBrowseAndRequest();
+  const { data: couponCodes } = useCouponCodes();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: enterpriseOffersData } = useEnterpriseOffers();
+
+  const learnerCreditSummaryCardData = useMemo(() => getLearnerCreditSummaryCardData({
+    enterpriseOffers: enterpriseOffersData.enterpriseOffers,
```

Comment:
While this refactor is parity the existing state, I do wonder if we should be relying on `enterpriseOffersData.currentEnterpriseOffers` to ensure we only consider non-expired offers within `getLearnerCreditSummaryCardData`.